universal-application-tool-0.0.1/app/services/applicant/ApplicantQuestion.java

```diff
@@ -582,6 +592,71 @@ public Path getNumberPath() {
     }
   }
 
+  public class MultiOptionQuestion implements PresentsErrors {
+
+    private Optional<ImmutableList<String>> selectedOptionsValue;
+
+    public MultiOptionQuestion() {
+      assertQuestionType();
+    }
+
+    @Override
+    public boolean hasQuestionErrors() {
+      return !getQuestionErrors().isEmpty();
+    }
+
+    public ImmutableSet<ValidationErrorMessage> getQuestionErrors() {
+      // TODO(https://github.com/seattle-uat/civiform/issues/416): Implement validation
+      return ImmutableSet.of();
+    }
+
+    @Override
+    public boolean hasTypeSpecificErrors() {
+      // There are no inherent requirements in a multi-option question.
+      return false;
+    }
+
+    public boolean hasValue() {
+      return getSelectedOptionsValue().isPresent();
+    }
+
+    public Optional<ImmutableList<String>> getSelectedOptionsValue() {
+      if (selectedOptionsValue != null) {
+        return selectedOptionsValue;
+      }
+
+      selectedOptionsValue = applicantData.readList(getSelectionPath());
```

Comment:
Question (blocking): Since we changed the scalar type, should this be updated as well?

Reply:
I think it makes sense to have a `SingleSelectQuestion` (dropdowns and radio buttons) and a `MultiSelectQuestion` (checkboxes). I updated to the new name and added a comment for implementing the other (linked to checkbox issue)